crates/ty_project/src/lib.rs

```diff
@@ -134,12 +141,18 @@ impl ProgressReporter for DummyReporter {
 impl Project {
     pub fn from_metadata(db: &dyn Db, metadata: ProjectMetadata) -> Result<Self, ToSettingsError> {
         let (settings, diagnostics) = metadata.options().to_settings(db, metadata.root())?;
-
-        let project = Project::builder(Box::new(metadata), Box::new(settings), diagnostics)
-            .durability(Durability::MEDIUM)
-            .open_fileset_durability(Durability::LOW)
-            .file_set_durability(Durability::LOW)
-            .new(db);
+        let check_mode = metadata.check_mode();
```

Comment:
Having `check_mode` on `ProjectMetadata` feels a bit odd. I think I'd prefer to default to `All` (You can implement `Default` for `CheckMode` and add the `#[default]` attribute to the field) and have the LSP explicitly set the check mode after startup.

Reply:
It would match my expectation to `Project` more closely. By default, I want to check all files. It's only the editor-like environments where we may want to deviate from this behavior (LSP, WASM). This also has the advantage that it doesn't require any additional setup code in the CLI (it does in the LSP and WASM but they opt into that by setting check mode to open files)